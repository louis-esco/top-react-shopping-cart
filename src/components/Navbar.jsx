import { NavLink } from "react-router-dom";

export default function Navbar() {
  return (
    <div className="navBar">
      <NavLink to={"/"}>Home</NavLink>
      <NavLink to={"/shop"}>Shop</NavLink>
      <NavLink to={"/cart"}>Cart</NavLink>
    </div>
  );
}
