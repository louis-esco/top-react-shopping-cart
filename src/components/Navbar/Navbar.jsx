import { NavLink } from "react-router-dom";
import styles from "./Navbar.module.css";
import PropTypes from "prop-types";

export default function Navbar({ getCartLength }) {
  const cartLength = getCartLength();
  return (
    <nav className={styles.navBar}>
      <NavLink
        className={(navData) =>
          navData.isActive
            ? `${styles.active} ${styles.navLink}`
            : styles.navLink
        }
        to={"/"}
      >
        HOME
      </NavLink>
      <NavLink
        className={(navData) =>
          navData.isActive
            ? `${styles.active} ${styles.navLink}`
            : styles.navLink
        }
        to={"/shop"}
      >
        SHOP
      </NavLink>
      <NavLink
        className={(navData) =>
          navData.isActive
            ? `${styles.active} ${styles.navLink}`
            : styles.navLink
        }
        to={"/cart"}
      >
        CART ({cartLength})
      </NavLink>
    </nav>
  );
}

Navbar.propTypes = {
  getCartLength: PropTypes.func,
};
