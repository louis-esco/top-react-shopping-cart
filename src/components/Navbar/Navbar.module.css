.navBar {
  display: flex;
  justify-content: center;
  gap: 150px;
  padding: 20px 0;
  background-color: #e76f51;
  position: sticky;
  top: 0px;
}

.navLink {
  text-decoration: none;
  color: #ffffff;
  font-size: 1.7rem;
  font-weight: 500;
}
