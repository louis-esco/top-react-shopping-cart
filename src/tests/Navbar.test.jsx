import { vi, describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "../components/Navbar/Navbar";

describe("Navbar component", () => {
  it("Renders Navbar", () => {
    const getCartLength = vi.fn();

    const { container } = render(
      <MemoryRouter>
        <Navbar getCartLength={getCartLength} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
