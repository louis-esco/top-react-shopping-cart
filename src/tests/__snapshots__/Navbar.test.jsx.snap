// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > Renders Navbar 1`] = `
<div>
  <nav
    class="_navBar_278c00"
  >
    <div
      class="_shopName_278c00"
    >
      <img
        alt="shop icon"
        src="/src/assets/icon.svg"
      />
      <div>
        Louis's shop
      </div>
    </div>
    <div
      class="_links_278c00"
    >
      <a
        aria-current="page"
        class="_active_278c00 _navLink_278c00"
        href="/"
      >
        HOME
      </a>
      <a
        class="_navLink_278c00"
        href="/shop"
      >
        SHOP
      </a>
      <a
        class="_navLink_278c00 _cart_278c00"
        href="/cart"
      >
        CART
        <div
          class="_cartLength_278c00"
        />
      </a>
    </div>
  </nav>
</div>
`;
